### PR TITLE
[SPRF-840] add base_date opts to Neurotech.compute_bacen_score/5

### PR DESCRIPTION
**Pull request by youalreadydid**
**Why is this change necessary?**

- To support the homologation environment of SCR Bacen.

**How does it address the issue?**

- Add base_date opts to Neurotech.compute_bacen_score/5.

**What side effects does this change have?**

- N/A

**Task card (link)**

- [SPRF-840](https://bcredi.atlassian.net/browse/SPRF-840)

### CODE REVIEW

File: lib/http_clients/neurotech.ex [hunk 1]
```diff
@@ -77,6 +82,14 @@ defmodule HttpClients.Neurotech do
     end
   end
 
+  defp put_base_date(inputs, base_date)
+  defp put_base_date(inputs, nil), do: inputs
+
+  defp put_base_date(inputs, base_date) do
```
Comment (mgiacomini) [inline lib/http_clients/neurotech.ex:88]:
```suggestion
  defp put_base_date(inputs, %Date{} = base_date) do
```

File: test/http_clients/neurotech_test.exs [hunk 2]
```diff
@@ -54,7 +54,12 @@ defmodule HttpClients.NeurotechTest do
   describe "compute_bacen_score/2" do
     test "returns score when the request succeeds" do
       response_body = bacen_response(:success)
-      mock(fn %{url: "/submit", method: :post} -> json(response_body) end)
+
+      mock(fn %{url: "/submit", method: :post, body: body} ->
+        assert not String.match?(body, ~r/PROP_BACEN_DATA_BASE/)
```
Comment (mgiacomini) [inline test/http_clients/neurotech_test.exs:59]:
```suggestion
```

Comment (mgiacomini) [reply test/http_clients/neurotech_test.exs:59]: isso não precisa ser testado aqui

File: test/http_clients/neurotech_test.exs [hunk 3]
```diff
@@ -67,6 +72,35 @@ defmodule HttpClients.NeurotechTest do
...
+      opts = [base_date: ~D[2017-10-01]]
+
+      assert {:ok, ^expected_analysis} =
+               Neurotech.compute_bacen_score(
+                 client(),
+                 credentials(),
+                 person,
+                 @transaction_id,
+                 opts
+               )
```
Comment (mgiacomini) [inline test/http_clients/neurotech_test.exs:101]:
```suggestion
      expected_analysis = %Score{
        score: 444,
        positive_analysis: "- Sem registro de vencidos no histórico.\r\n",
        negative_analysis: "- LIMITE DE CRÉDITO abaixo de R$1.000,00 no histórico.\r\n"
      }

      person = %Neurotech.Person{cpf: "65661563051"}
      opts = [base_date: ~D[2017-10-01]]

      assert {:ok, ^expected_analysis} =
               Neurotech.compute_bacen_score(
                 client(),
                 credentials(),
                 person,
                 @transaction_id,
                 opts
               )
```

File: lib/http_clients/neurotech.ex (hunk 1 above)
Comment (youalreadydid) [reply lib/http_clients/neurotech.ex]: feito!

File: test/http_clients/neurotech_test.exs (hunk 2 above)
Comment (youalreadydid) [reply test/http_clients/neurotech_test.exs]: feito!

File: test/http_clients/neurotech_test.exs (hunk 3 above)
Comment (youalreadydid) [reply test/http_clients/neurotech_test.exs]: @mgiacomini feito!

**Review by adrianosiq — APPROVED** [review]

**Review by Gabee01 — APPROVED** [review]

**Pull request merged**